packages/sanity/src/core/bundles/hooks/useDocumentVersions.tsx

```diff
@@ -37,13 +37,13 @@ export function useDocumentVersions(props: DocumentPerspectiveProps): DocumentPe
   const {documentId} = props
 
   const {data: bundles} = useBundles()
-  const publishedId = getPublishedId(documentId, documentId.includes('.'))
+  const publishedId = getPublishedId(documentId)
 
   const documentPreviewStore = useDocumentPreviewStore()
 
   const observable = useMemo(() => {
     return documentPreviewStore
-      .unstable_observeDocumentIdSet(`sanity::versionOf("${publishedId}")`)
+      .unstable_observeDocumentIdSet(`_id in path("versions.**") && _id match '*.${publishedId}'`)
```

Comment:
Maybe add a TODO comment here to make this proper? It's in the commit, but also nice to have it in code

Reply:
My hope was that CL had released the updated versionOf before this PR was ready (it was meant to be on Thursday but they had some delays), so I had expected this code to be reverted (need to fix the commit message 🥲) 

I'll add the todo on the code 👍